ambry-commons/src/main/java/com.github.ambry.commons/ByteBufferReadableStreamChannel.java

```diff
@@ -65,6 +74,30 @@ public long getSize() {
   }
 
   @Override
+  public void setDigestAlgorithm(String digestAlgorithm)
+      throws NoSuchAlgorithmException {
+    if (digest != null && digest.getAlgorithm().equals(digestAlgorithm)) {
```

Comment:
I have added docs to describe this. Since I have already written this and its not doing any harm and just makes the class more versatile, I think we can leave it as is. It will take some state maintenance to achieve ordering too and I don't think this code is more complicated than code that will be required for state maintenance.